src/cloud_provider/aws/kubernetes/mod.rs

```
@@ -547,6 +560,106 @@ impl EKS {
         Ok(context)
     }
 
+    // Return a pair of:
+    // update_desired_node: bool
+    // desired_nodes_count: i32
+    fn should_update_desired_nodes(&self, event_details: EventDetails) -> Result<(bool, i32), EngineError> {
+        let future_node_group = match self.nodes_groups.is_empty() {
+            false => self.nodes_groups.first().unwrap(),
+            true => {
+                return Err(EngineError::new_cluster_has_no_worker_nodes(
+                    event_details,
+                    Option::Some(CommandError::new_from_safe_message(
+                        "Could not find node_group in terra context".to_string(),
+                    )),
+                ))
+            }
+        };
+
+        let region = RusotoRegion::from_str(&self.region())
+            .unwrap_or_else(|_| panic!("S3 region `{}` doesn't seems to be valid.", self.region.to_aws_format()));
```

Should not fire a panic here, please return an Engine Error